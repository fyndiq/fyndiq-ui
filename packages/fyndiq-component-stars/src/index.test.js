import React from 'react'
import { shallow } from 'enzyme'

import Stars from './'

describe('fyndiq-component-stars', () => {
  it('should have a default onChange handler', () => {
    expect(Stars.defaultProps.onChange(3)).toBe(3)
  })
  it('should be rendered without props', () => {
    const component = shallow(<Stars />)
    expect(component).toMatchSnapshot()
  })

  it('should be rendered with prop rating', () => {
    const component = shallow(<Stars rating={4.5} />)
    expect(component).toMatchSnapshot()
  })

  it('should be rendered with prop reviews', () => {
    const component = shallow(<Stars reviews={4} />)
    expect(component).toMatchSnapshot()
  })

  it('should be rendered with prop size', () => {
    const component = shallow(<Stars size="l" />)
    expect(component).toMatchSnapshot()
  })

  it('should be rendered with prop reviews', () => {
    const component = shallow(<Stars interactive />)
    expect(component).toMatchSnapshot()
  })

  it('should send an event if interactive and clicked on', () => {
    const clickSpy = jest.fn()
    const component = shallow(<Stars interactive onChange={clickSpy} />)

    // Find the 3rd star (at is 0-index based)
    component.find('Star').at(2).simulate('click')

    expect(clickSpy.mock.calls).toHaveLength(1)
    expect(clickSpy.mock.calls[0][0]).toBe(3)
  })

  it('should change its state when hovered and interactive', () => {
    const component = shallow(<Stars interactive />)
    component.find('Star').at(2).simulate('hover')
    expect(component).toMatchSnapshot()
  })

  it('should not change its state when hovered and not interactive', () => {
    const component = shallow(<Stars rating={4} />)
    component.find('Star').at(2).simulate('hover')
    expect(component).toMatchSnapshot()
  })

  it('should reset its state when the mouse leaves', () => {
    const component = shallow(<Stars rating={4.3} interactive />)
    component.find('Star').at(2).simulate('hover')
    component.find('.stars').simulate('mouseleave')
    expect(component).toMatchSnapshot()
  })
})
