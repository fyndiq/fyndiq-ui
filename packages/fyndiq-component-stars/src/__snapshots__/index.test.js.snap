// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-stars should be rendered with prop rating 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0.5}
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should be rendered with prop reviews 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
  <span
    className="reviews"
  >
    (
    4
     reviews)
  </span>
</div>
`;

exports[`fyndiq-component-stars should be rendered with prop reviews 2`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              interactive
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should be rendered with prop size 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-l
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should be rendered without props 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={NaN}
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should change its state when hovered and interactive 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              interactive
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0}
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should not change its state when hovered and not interactive 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0}
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;

exports[`fyndiq-component-stars should reset its state when the mouse leaves 1`] = `
<div
  className="wrapper"
>
  <div
    className="
              stars
              interactive
              size-m
            "
    data-test="STARS"
    onMouseLeave={[Function]}
  >
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={1}
      onClick={[Function]}
      onHover={[Function]}
    />
    <Star
      full={0.2999999999999998}
      onClick={[Function]}
      onHover={[Function]}
    />
  </div>
</div>
`;
